src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -50,50 +59,81 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
           return isValidSharePointUrl;
         }
 
-        const invalidIds = formatting
-          .splitAndTrim(args.options.ids)
-          .filter(id => !validation.isValidGuid(id));
-        if (invalidIds.length > 0) {
-          return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+        if (args.options.ids) {
+          const invalidIds = formatting
+            .splitAndTrim(args.options.ids)
+            .filter(id => !validation.isValidGuid(id));
+
+          if (invalidIds.length > 0) {
+            return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+          }
         }
 
         return true;
       }
     );
   }
 
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['ids', 'allPrimary', 'allSecondary'] });
+  }
+
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     if (this.verbose) {
       await logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
     }
 
-    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
-
-    const ids: string[] = formatting.splitAndTrim(args.options.ids);
-    const idsChunks: string[][] = [];
+    let requestUrl: string = `${args.options.siteUrl}/_api`;
 
-    while (ids.length) {
-      idsChunks.push(ids.splice(0, 20));
+    if (args.options.ids) {
+      requestUrl += `/site/RecycleBin/RestoreByIds`;
+    }
+    else if (args.options.allPrimary) {
+      requestUrl += `/web/RecycleBin/RestoreAll`;
+    }
+    else if (args.options.allSecondary) {
+      requestUrl += `/site/RecycleBin/RestoreAll`;
```

Comment:
I noticed that when we empty the second-stage recycle bin, we are also restoring the first-stage recycle bin.

Reply:
Hi @milanholemans 
Sorry, I did not get this point. Can you please explain?

Reply:
I noticed that when you have items in your 1st-stage and 2nd-stage recycle bin, and you call this API, the whole recycle bin is restored. So both the first stage and 2nd-stage.

Reply:
Hi @milanholemans 
I get your point. However, there doesn't seem to be any other API available to restore only 2nd-stage recycle bin.
In this case, should we add this point as a Remark in the documentation?

Reply:
I haven't really had the time to do some research myself, but according to me, this is not the wanted behavior that the user wants. Worst case we have to fetch all items in the second-stage recycle bin and restore them by ID (in batch).

@pnp/cli-for-microsoft-365-maintainers what is your opinion on this?
TL;DR: We're working on the command [spo site recyclebinitem restore](https://pnp.github.io/cli-microsoft365/cmd/spo/site/site-recyclebinitem-restore) where we'll add 2 new options `allPrimary` and `allSecondary`.
There are 2 nice API endpoints for this:

**Restore all items from first-stage:**

```http
POST https://tenant.sharepoint.com/sites/project-x/_api/web/RecycleBin/RestoreAll
```

**Restore all items from second-stage:**

```http
POST https://tenant.sharepoint.com/sites/project-x/_api/site/RecycleBin/RestoreAll
```

The API request to restore the first-stage recycle bin is working fine. However, I noticed when we use the second-stage equivalent, that both first-stage **and** second-stage recycle bin items are stored.
According to me if you specify `allSecondy`, I also expect to only restore the second-stage recycle bin and leave the first-stage as is.

Reply:
@nanddeepn, I've been playing around a bit and found something that might work:

```http
POST https://tenant.sharepoint.com/sites/Project-x/_api/site/GetRecyclebinItems(rowLimit=2000000000,itemState=2)/RestoreAll
```

Reply:
Hi @milanholemans 
Yes, this API endpoint works perfectly. Thanks!
Updated the PR.

Reply:
Nice find @milanholemans!